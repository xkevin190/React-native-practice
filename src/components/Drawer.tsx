import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import React, { useEffect } from 'react';
import {Text, StyleSheet, SafeAreaView} from 'react-native';
import {TouchableOpacity} from 'react-native-gesture-handler';
import Icon from 'react-native-vector-icons/Ionicons';
import {colors} from '../constants/colors';
import {normalize, SCREEN_WIDTH} from '../constants/utils';

interface DrawerProps {
  navigation: NativeStackNavigationProp<any, any>;
}

const Drawer = ({navigation}: DrawerProps) => {
  const back = () => navigation.closeDrawer();

  const navigate = (name: string) => {
    navigation.navigate(name);
  };

  return (
    <SafeAreaView style={styles.container}>
      <TouchableOpacity onPress={back}>
        <Icon
          style={styles.iconStyle}
          name={'close'}
          size={normalize(25)}
          color={colors.primary}
        />
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.itemContainer}
        onPress={() => navigate('Home')}>
        <Text style={styles.textItem}>Characters</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.itemContainer}
        onPress={() => navigate('Favorites')}>
        <Text style={styles.textItem}>Favorites</Text>
      </TouchableOpacity>
    </SafeAreaView>
  );
};

export default Drawer;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.background,
  },
  itemContainer: {
    marginTop: 40,
    width: '100%',
    marginHorizontal: 45,
  },
  textItem: {
    fontFamily: 'Urbanist',
    fontSize: normalize(20),
    fontWeight: '600',
    color: colors.primary,
  },
  iconStyle: {
    marginBottom: 10,
    marginHorizontal: 20,
  },
});
