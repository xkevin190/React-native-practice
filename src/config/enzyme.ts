import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

/**
 * @tech-debt
 *
 * Update enzyme to support react 18
 */
Enzyme.configure({ adapter: new Adapter() });
