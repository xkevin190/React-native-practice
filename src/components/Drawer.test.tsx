import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import Drawer from './Drawer';
import {View} from 'react-native';

describe('Drawer (Unit-Test)', () => {
  let wrapper: ShallowWrapper;

  const navigation = {
    navigate: jest.fn(),
    closeDrawer: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<Drawer navigation={navigation} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('closeDrawer function should executed', () => {
    wrapper.find('TouchableOpacity').first().simulate('press');
    expect(navigation.closeDrawer).toBeCalled();
  });

  it(' should navigate to "Characters" screen', () => {
    console.log(wrapper.debug());
    wrapper.find('TouchableOpacity').at(1).simulate('press');
    expect(navigation.navigate).toBeCalledWith('Home');
  });

  it(' should navigate to "Favorites" screen', () => {
    console.log(wrapper.debug());
    wrapper.find('TouchableOpacity').at(2).simulate('press');
    expect(navigation.navigate).toBeCalledWith('Favorites');
  });
});
