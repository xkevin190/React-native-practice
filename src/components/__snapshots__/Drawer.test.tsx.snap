// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer (Unit-Test) should match snapshot 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#17141f",
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    delayLongPress={600}
    extraButtonProps={
      Object {
        "exclusive": true,
        "rippleColor": "transparent",
      }
    }
    onPress={[Function]}
  >
    <Icon
      allowFontScaling={false}
      color="#ffffff"
      name="close"
      size={57}
      style={
        Object {
          "marginBottom": 10,
          "marginHorizontal": 20,
        }
      }
    />
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    delayLongPress={600}
    extraButtonProps={
      Object {
        "exclusive": true,
        "rippleColor": "transparent",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginHorizontal": 45,
        "marginTop": 40,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Urbanist",
          "fontSize": 45,
          "fontWeight": "600",
        }
      }
    >
      Characters
    </Text>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    delayLongPress={600}
    extraButtonProps={
      Object {
        "exclusive": true,
        "rippleColor": "transparent",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "marginHorizontal": 45,
        "marginTop": 40,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Urbanist",
          "fontSize": 45,
          "fontWeight": "600",
        }
      }
    >
      Favorites
    </Text>
  </TouchableOpacity>
</ForwardRef(SafeAreaView)>
`;
